lib/db.js

```diff
@@ -0,0 +1,182 @@
+import knex from 'knex';
+import log from './logger';
+import { DataStream } from 'scramjet';
+import { promisify } from 'util';
+import moment from 'moment';
+
+let _pg = null;
+function getDB () {
+  let knexConfig;
+  _pg = _pg || (async () => {
+    try {
+      knexConfig = require('../knexfile-local');
+      log.info('Using connection data from knexfile-local');
+    } catch (e) {
+      knexConfig = require('../knexfile');
+      log.info('Using connection data from production knexfile');
+    }
+
+    const pg = knex(knexConfig);
+    log.info('Connecting to database.');
+
+    return pg;
+  })()
+    .then(async pg => {
+      await pg.migrate.latest(knexConfig);
+      log.info('Database connected and migrations are handled, ready to roll!');
+      return pg;
+    });
+  // Run any needed migrations and away we go
+  return _pg;
+}
+
+/**
+ * Build an object that can be inserted into our database.
+ * @param {object} m measurement object
+ * @return {object} an object capable of being saved into the PostgreSQL database
+ */
+function convertMeasurementToSQLObject (stream, pg) {
```

Comment:
Trying the latest, when running with command below, I'm getting an error.

```
$ node index.js -s 'StateAir_Hanoi' -d                                                            [±:signicode-feature/streamed-libs:∂]
2018-09-26T13:41:29.746Z - info: --- Dry run for Testing, nothing is saved to the database. ---
2018-09-26T13:41:29.775Z - info: Looking up source StateAir_Hanoi
2018-09-26T13:41:30.276Z - info: Getting stream for "StateAir_Hanoi" from "stateair"
2018-09-26T13:41:30.685Z - info: Using connection data from knexfile-local
2018-09-26T13:41:30.756Z - info: Connecting to database.
2018-09-26T13:41:30.958Z - info: Database connected and migrations are handled, ready to roll!
2018-09-26T13:41:31.164Z - error: Runtime error occurred in DataStream(unused): Error: Cannot read property 'utc' of undefined
    at Promise.all.catch (/Users/jflasher/workspace/openaq-fetch/node_modules/scramjet-core/lib/util/mk-transform.js:59:44)
    at <anonymous>
  caused by:
    at DataStream.stream.map.data (/Users/jflasher/workspace/openaq-fetch/lib/db.js:50:17)
    at <anonymous>
  --- raised in DataStream(unused) constructed ---
    at PromiseTransformStream (/Users/jflasher/workspace/openaq-fetch/node_modules/scramjet-core/lib/util/promise-transform-stream.js:65:27)
    at DataStream (/Users/jflasher/workspace/openaq-fetch/node_modules/scramjet-core/lib/data-stream.js:55:9)
    at DataStream.map (/Users/jflasher/workspace/openaq-fetch/node_modules/scramjet-core/lib/data-stream.js:153:26)
    at convertMeasurementToSQLObject (/Users/jflasher/workspace/openaq-fetch/lib/db.js:41:17)
    at DataStream.use (/Users/jflasher/workspace/openaq-fetch/node_modules/scramjet-core/lib/data-stream.js:304:20)
    at streamRecordsToPg (/Users/jflasher/workspace/openaq-fetch/lib/db.js:75:6)
    at DataStream.use (/Users/jflasher/workspace/openaq-fetch/node_modules/scramjet-core/lib/data-stream.js:304:20)
    at streamDataToDB (/Users/jflasher/workspace/openaq-fetch/lib/db.js:93:6)
    at /Users/jflasher/workspace/openaq-fetch/lib/db.js:178:9
    at <anonymous>
```

I put a log statement to see what the `data` variable was getting. In the case where it errors, it's getting the below which looks like the already converted object. So it looks like this function may be getting called multiple times and will fail if it runs on already converted data, which I suppose is good! 

Note also that the coordinates on the converted object is the full postgis coordinates object, but just needs to be a simple lat/lon object.

```
{ location: 'US Diplomatic Post: Hanoi',
  value: 17,
  unit: 'µg/m³',
  parameter: 'pm25',
  country: 'VN',
  city: 'Hanoi',
  source_name: 'StateAir_Hanoi',
  date_utc: '2018-09-26T12:00:00.000Z',
  source_type: 'government',
  mobile: false,
  data: 
   { date: 
      { utc: '2018-09-26T12:00:00.000Z',
        local: '2018-09-26T19:00:00+07:00' },
     parameter: 'pm25',
     value: 17,
     unit: 'µg/m³',
     averagingPeriod: { value: 1, unit: 'hours' },
     location: 'US Diplomatic Post: Hanoi',
     city: 'Hanoi',
     country: 'VN',
     coordinates: { latitude: 21.021938, longitude: 105.81881 },
     attribution: [ [Object] ],
     sourceName: 'StateAir_Hanoi',
     sourceType: 'government',
     mobile: false },
  coordinates: 
   Raw {
     client: 
      Client_PG {
        config: [Object],
        logger: [Object],
        connectionSettings: [Object],
        driver: [Object],
        pool: [Object],
        valueForUndefined: [Object],
        _events: [Object],
        _eventsCount: 4,
        makeKnex: [Function: makeKnex],
        version: '9.5.3' },
     sql: 'ST_geomFromText(?, ?)',
     bindings: [ 'Point(105.81881 21.021938)', 4326 ],
     _wrappedBefore: undefined,
     _wrappedAfter: undefined,
     _debug: undefined } }
```

Reply:
I'll check it today or tomorrow evening (should be fixed tomorrow your time).

Reply:
Oh, now I see my misunderstaning. It's the `-d` - a small fix resolved this.

Reply:
So this was due to the fact that I removed an unneeded argument from one of the methods and didn't change the call. Now this will work as it should.

Reply:
Just dropped the `-d` and it's still happening, same error.

Reply:
I found it now. It didn't occur if the data was not streamed to S3. As you suspected the object was transformed to the SQL version twice.

Reply:
Hi @MichalCz, I wanted to make sure I should be waiting for another commit before checking fetch behavior? I didn't see any commit since your last comment and wanted to make sure I didn't miss something. Thanks!

Reply:
Hi @jflasher I didn't push to the right remote after I fixed the issue. :(

There's that commit now.

Reply:
Great, this looks to be working now, thanks!